core/src/data/tileSource.cpp

```diff
@@ -98,7 +98,7 @@ void TileSource::loadTileData(std::shared_ptr<TileTask> _task, TileTaskCb _cb) {
     }
 
     for (auto& subTask : _task->subTasks()) {
-        subTask->source().loadTileData(subTask, _cb);
+        subTask->source()->loadTileData(subTask, _cb);
```

Comment:
Same, check for validity of the source here. Though I think it might be unlikely that subtask source is null and main task's source is not, in which case this will never be called, maybe put an assert in that case?

Reply:
This is called within `Map::update()` only for TileSources of the current Scene - so this should be safe.